frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -200,35 +207,14 @@ function ReaderStatus:deleteFile()
     FileManager:showDeleteFileDialog(self.document.file, post_delete_callback, pre_delete_callback)
 end
 
-function ReaderStatus:onShowBookStatus(before_show_callback)
-    local status_page = BookStatusWidget:new {
-        thumbnail = FileManagerBookInfo:getCoverImage(self.document),
-        props = self.ui.doc_props,
-        document = self.document,
-        settings = self.settings,
-        ui = self.ui,
-    }
-    if before_show_callback then
-        before_show_callback()
-    end
-    status_page.dithered = true
-    UIManager:show(status_page, "full")
-    return true
-end
-
 -- If mark_read is true then we change status only from reading/abandoned to complete.
 -- Otherwise we change status from reading/abandoned to complete or from complete to reading.
-function ReaderStatus:onMarkBook(mark_read)
-    self.summary.status = (not mark_read and self.summary.status == "complete") and "reading" or "complete"
-    self.summary.modified = os.date("%Y-%m-%d", os.time())
-    -- If History is called over Reader, it will read the file to get the book status, so save and flush
-    self.settings:saveSetting("summary", self.summary)
-    self.settings:flush()
-end
-
-function ReaderStatus:onReadSettings(config)
-    self.settings = config
-    self.summary = config:readSetting("summary") or {}
+function ReaderStatus:markBook(mark_read)
```

Comment:
Any technical reason to change this from an "on" event handler to just a method ?
Or it's just you wanting to make more things private and less candidate for abuse?

Reply:
This method is **not** called by event from anywhere, do not know why it got the "on" prefix.

Reply:
Doesn't that imply setting book status is missing from Dispatcher? I don't know if that matters though. :-)